File: app/src/main/java/org/medhis2yes/usescases/programEventDetail/ProgramEventDetailComponent.java
package org.medhis2yes.usescases.programEventDetail;

import org.medhis2yes.data.dagger.PerActivity;
import org.medhis2yes.usescases.programEventDetail.eventList.EventListComponent;
import org.medhis2yes.usescases.programEventDetail.eventList.EventListModule;
import org.medhis2yes.usescases.programEventDetail.eventMap.EventMapComponent;
import org.medhis2yes.usescases.programEventDetail.eventMap.EventMapModule;

import dagger.Subcomponent;

@PerActivity
@Subcomponent(modules = ProgramEventDetailModule.class)
public interface ProgramEventDetailComponent {
    void inject(ProgramEventDetailActivity activity);

    EventListComponent plus(EventListModule eventListModule);

    EventMapComponent plus(EventMapModule eventMapModule);
}

File: app/src/main/java/org/dhis2_haparent/usescases/programEventDetail/ProgramEventDetailComponent.kt
package org.dhis2_haparent.usescases.programEventDetail

import dagger.Subcomponent
import org.dhis2_haparent.commons.di.dagger.PerActivity
import org.dhis2_haparent.usescases.programEventDetail.eventList.EventListComponent
import org.dhis2_haparent.usescases.programEventDetail.eventList.EventListModule
import org.dhis2_haparent.usescases.programEventDetail.eventMap.EventMapComponent
import org.dhis2_haparent.usescases.programEventDetail.eventMap.EventMapModule

@PerActivity
@Subcomponent(modules = [ProgramEventDetailModule::class])
interface ProgramEventDetailComponent {
    fun inject(activity: ProgramEventDetailActivity)
    operator fun plus(eventListModule: EventListModule): EventListComponent
    operator fun plus(eventMapModule: EventMapModule): EventMapComponent
}
